packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -113,10 +115,22 @@ class Observatory {
     });
   }
 
-  Future<Response> isolateReload(String isolateId) {
-    return sendRequest('isolateReload', <String, dynamic>{
-      'isolateId': isolateId
-    });
+  Future<bool> reloadSources(String isolateId) async {
+    Completer<Event> whenIsolateReloads = new Completer<Event>();
+    StreamSubscription<Event> sub = onIsolateEvent
+      .where((Event event) => event.kind == 'IsolateReload')
+      .listen((Event event) => whenIsolateReloads.complete(event));
```

Comment:
nit: you shouldn't need to define a closure here, as `whenIsolateReloads.complete` tear-off satisfies the signature needed by `onData`.